packages/shared/src/archive-storybook/index.ts

```diff
@@ -22,7 +21,14 @@ export function buildArchiveStorybook(
 ) {
   checkArchivesDirExists(defaultOutputDir);
   addViewportsToStoriesFiles(defaultOutputDir).then(() => {
-    const binPath = resolve(dirname(require.resolve('storybook/package.json')), './index.js');
-    execFileSync('node', [binPath, 'build', ...processArgs, '-c', configDir], { stdio: 'inherit' });
+    execFileSync('node', [binPath(), 'build', ...processArgs, '-c', configDir], {
+      stdio: 'inherit',
+    });
   });
 }
+
+function binPath() {
+  // eslint-disable-next-line global-require
+  const packageJson = require('storybook/package.json');
+  return resolve(dirname(require.resolve('storybook/package.json')), packageJson.bin.storybook);
```

Comment:
There might be some windows path issue here, I'm not 100% sure.
I think it will be fine, considering how the code was before.

But the reason I raise it is because the path for the bin (`packageJson.bin.storybook`), will always be POSIX (linux).
Where as `dirname`, `require.resolve` and `resolve` will use platform dependent paths.

See:
https://github.com/storybookjs/storybook/blob/cbac722b410129b155b1886326b1d3274309b423/code/lib/cli/package.json#L277

I suspect `path.resolve` will work with the combined input, but I'm just not 1000% sure.
And I do not have a windows machine to test it on.

chatGPT didn't give me a conclusive answer :(

Perhaps this is a good reason/moment to add a CI test on a windows platform?

Reply:
Considering this was the code before:
```ts
const binPath = resolve(dirname(require.resolve('storybook/package.json')), './index.js');
```
The above question shouldn't be a blocker to get this merged IMHO.

Reply:
Thanks a bunch @ndelangen!